crates/e2e-testing/src/spin_controller.rs

```diff
@@ -46,6 +49,22 @@ impl Controller for SpinUp {
     ) -> Result<AppInstance> {
         let appdir = spin::appdir(app_name);
 
+        //cleanup state dir before spin up
+        let state_dir = match env::var("STATE_DIR") {
```

Comment:
This could accidentally delete a "real" state dir if a user happens to have a full path in their env. Maybe:

```suggestion
        let state_dir = match env::var("TEST_STATE_DIR") {
```

Reply:
this is a good point. (and a one that you had mentioned to me earlier about something similar, so my bad for not remembering it). thanks Lann.

i have removed the use of STATE_DIR